### DISCUSSION TITLE
Can't listen to event fired from a child component after wire:navigate to parent page from the parent page

### DISCUSSION THREAD

**Discussion by m7moudabdel7mid** [livewire/livewire#6099] [7. Bugs (archived)]
### Livewire version

3.0@beta

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Hi,

Here is an example where i have a parent component which has p tag that displays the value of public attribute $text.
The child component has a text input, when updated an event is triggered with the text from the input.
The parent listens to that event and change the text property to the one from the event.

ParentComponent class
```
public $text = '';

#[On('my-event')]
public function change_text($text)
{
    $this->text = $text;
}
```

parent-component view
```
<div>
    <a href="{{ route('parent') }}" wire:navigate>Go Home</a>

    <p>{{ $text }}</p>

    <livewire:child-component />
</div>
```

ChildComponent class
```
public $text = '';

public function updated()
{
    $this->dispatch('my-event', text: $this->text);
}
```

child-component view
```
<div>
    <input type="text" wire:model.live="text">
</div>
```

the parent component's full page route
```
Route::get('/', ParentComponent::class)->name('parent');
```

Now when you open the parent page, click the "Go Home" link to navigate to the parent page again and when you update the input value you will get an error in the browser console `Uncaught (in promise) undefined` and the $test value won't be updated.

### Additional information

a practical example of this issue is that the user might be in a page and open the same page from the header or sidebar, maybe to see if there is any new updates or something

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Are you using the latest beta release ([v3.0.0-beta.5](https://github.com/livewire/livewire/releases/tag/v3.0.0-beta.5))?

**Reply by m7moudabdel7mid** [reply]
Actually i was using v3.0.0-beta.4 but now i upgraded to v3.0.0-beta.5 and the issue still exists.

**Comment by joshhanley** [comment]
I submitted a failing test PR #6264 which has been fixed and merged. So this will be in the next release. 🙂
